src/aioftp/client.py

```diff
@@ -566,11 +704,12 @@ def parse_mlsx_line(self, b):
             s = b
         line = s.rstrip()
         facts_found, _, name = line.partition(" ")
-        entry = {}
+        entry: Dict[str, str] = {}
         for fact in facts_found[:-1].split(";"):
             key, _, value = fact.partition("=")
             entry[key.lower()] = value
-        return pathlib.PurePosixPath(name), entry
+
+        return pathlib.PurePosixPath(name), cast(InfoDict, entry)
```

Comment:
It is impossible to type every part of the code because python is very dynamic. It's also impossible to do without cast. The input is just bytes/strings and whether they are cast to the desired structure depends on who uses the function.

Reply:
In the other case here, you have to do some validation on a few lines and throw an error.

Reply:
In this case you can add a `MLSxDict` or something like that, which will use only required for further code keys (like `"type"`) and type checker will help here, by the way. And after all `InfoDict` will be a union of three (or even more) type dicts.

Reply:
So, here I don't really understand what you're suggesting.

Reply:
I suggest to add `MLSxDict` with required fields and add it to `InfoDict` union.